contributors/design-proposals/add-on/coredump-daemonset.md

```diff
@@ -0,0 +1,117 @@
+# coredump detector add-on
+
+Cao Shufeng (@CaoShuFeng)
+
+## Abstract
+
+This proposal aims at managing coredump files generated in cluster containers.
+
+## Motivation and Goals
+
+Application coredump is one of required feature for enterprise systems. In Kubernetes, we need to take "multi-tenancy" and "security" into account. Technically, under Linux, each server’s coredump can be controlled via [/proc/sys/kernel/core_pattern](http://man7.org/linux/man-pages/man5/core.5.html) of the host. By this, an application coredump can be passed into host’s special file or special program. We need to build API to specify
+  - Where coredumps should be stored and access control to them
+  - typically, namespaced persistent storage rather than host’s file system
+  - list/get/delete coredump metadata
+  - access control to dumped data
+  - quota for coredump for each namespace
+  - no core dump label for specific environment (such as development environ)
+
+## Non-Goals
+This project amis to support access control to coredump files generated from local Linux namespace isolated container runtimes, like docker, rkt. Hypervisor-based container runtime like frakti and container runtimes use other implementations for pod isolation are not considered.
+
+## Community Work
+
+- make core_pattern support namespace in kernel (not approved yet): https://lkml.org/lkml/2017/2/7/931
+
+## Proposed Design
+
+To implement this, we need to make the following components run in kubernetes cluster:
+1. A coredump-detector daemonset runs in each node. It will:
+   * put a new special program integrated with kuberentes’s API into /proc/sys/kernel/core_pattern
```

Comment:
For OSes using systemd, I don't see the value in replacing the systemd default.  Instead, coredumps can be gathered from it.